sdcm/sct_events/database.py

```diff
@@ -380,17 +384,19 @@ def get_pattern_to_event_to_func_mapping(node: str) \
     mapping = []
     event_registry = ContinuousEventsRegistry()
 
-    def _add_event(event_type: Type[ScyllaDatabaseContinuousEvent], match: Match):
+    def _add_event(event_type: Type[ScyllaDatabaseContinuousEvent], match: Match, publish: bool = True):
         kwargs = match.groupdict()
         if "shard" in kwargs:
             kwargs["shard"] = int(kwargs["shard"])
-        event_type(node=node, **kwargs).begin_event()
+        event_type(node=node, **kwargs).begin_event(publish=publish)
 
-    def _end_event(event_type: Type[ScyllaDatabaseContinuousEvent], match: Match):
+    def _end_event(event_type: Type[ScyllaDatabaseContinuousEvent], match: Match, publish: bool = True):
         kwargs = match.groupdict()
         continuous_hash = event_type.get_continuous_hash_from_dict({'node': node, **kwargs})
         if begin_event := event_registry.find_continuous_events_by_hash(continuous_hash):
-            begin_event[-1].end_event()
+            if match.group() == event_type.failure_string:
```

Comment:
I would argue, this probably should be a regex, and not testing it by equability. so in cases we have a bit more sophisticate log print to identify